bundles/org.openhab.binding.teslascope/src/main/java/org/openhab/binding/teslascope/internal/TeslascopeHandler.java

```diff
@@ -0,0 +1,381 @@
+        updateState(TeslascopeBindingConstants.CHANNEL_SOFTWARE_UPDATE_VERSION,
+                new StringType(detailedInformation.softwareUpdateVersion));
+        updateState(TeslascopeBindingConstants.CHANNEL_SUNROOF_STATE, new StringType(detailedInformation.sunRoofState));
+        updateState(TeslascopeBindingConstants.CHANNEL_SUNROOF,
+                new DecimalType(detailedInformation.sunRoofPercentOpen));
+        updateState(TeslascopeBindingConstants.CHANNEL_HOMELINK, OnOffType.from(detailedInformation.homelinkNearby));
+        updateState(TeslascopeBindingConstants.CHANNEL_TPMS_FL,
+                new QuantityType<>(detailedInformation.tpmsPressureFL, Units.BAR));
+        updateState(TeslascopeBindingConstants.CHANNEL_TPMS_FR,
+                new QuantityType<>(detailedInformation.tpmsPressureFR, Units.BAR));
+        updateState(TeslascopeBindingConstants.CHANNEL_TPMS_RL,
+                new QuantityType<>(detailedInformation.tpmsPressureRL, Units.BAR));
+        updateState(TeslascopeBindingConstants.CHANNEL_TPMS_RR,
+                new QuantityType<>(detailedInformation.tpmsPressureRR, Units.BAR));
+        updateState(TeslascopeBindingConstants.CHANNEL_TPMS_SOFT_WARNING_FL,
+                OnOffType.from(detailedInformation.tpmsSoftWarningFL));
+        updateState(TeslascopeBindingConstants.CHANNEL_TPMS_SOFT_WARNING_FR,
+                OnOffType.from(detailedInformation.tpmsSoftWarningFR));
+        updateState(TeslascopeBindingConstants.CHANNEL_TPMS_SOFT_WARNING_RL,
+                OnOffType.from(detailedInformation.tpmsSoftWarningRL));
+        updateState(TeslascopeBindingConstants.CHANNEL_TPMS_SOFT_WARNING_RR,
+                OnOffType.from(detailedInformation.tpmsSoftWarningRR));
+        updateState(TeslascopeBindingConstants.CHANNEL_DRIVER_FRONT_DOOR,
+                detailedInformation.df ? OpenClosedType.OPEN : OpenClosedType.CLOSED);
+        updateState(TeslascopeBindingConstants.CHANNEL_DRIVER_REAR_DOOR,
+                detailedInformation.dr ? OpenClosedType.OPEN : OpenClosedType.CLOSED);
+        updateState(TeslascopeBindingConstants.CHANNEL_PASSENGER_FRONT_DOOR,
+                detailedInformation.pf ? OpenClosedType.OPEN : OpenClosedType.CLOSED);
+        updateState(TeslascopeBindingConstants.CHANNEL_PASSENGER_REAR_DOOR,
+                detailedInformation.pr ? OpenClosedType.OPEN : OpenClosedType.CLOSED);
+        updateState(TeslascopeBindingConstants.CHANNEL_FRONT_TRUNK, OnOffType.from(detailedInformation.ft));
+        updateState(TeslascopeBindingConstants.CHANNEL_REAR_TRUNK, OnOffType.from(detailedInformation.rt));
+
+        // virtual items
+        updateState(TeslascopeBindingConstants.CHANNEL_HONK_HORN, OnOffType.OFF);
+        updateState(TeslascopeBindingConstants.CHANNEL_FLASH_LIGHTS, OnOffType.OFF);
+    }
+
+    protected void ac(boolean b) throws TeslascopeCommunicationException, TeslascopeAuthenticationException {
```

Comment:
```suggestion
    protected void setACPower(boolean b) throws TeslascopeCommunicationException, TeslascopeAuthenticationException {
```

Reply:
Updated to setAutoConditioning

Reply:
Not `setAirConditioning` ?

Reply:
no - the control on the Tesla side is auto-conditioning - this pre-conditions the car interior based on the ambient conditions - so will trigger not just the AC,  but also things like seat/steering wheel heaters if it's cold enough (like it is right in Australia :) )